boards/common/nrf52/Kconfig

```diff
@@ -9,3 +9,31 @@ config BOARD_COMMON_NRF52
     select HAS_PERIPH_RTT
     select HAS_PERIPH_TIMER
     select HAS_RIOTBOOT
+
+if TEST_KCONFIG
+
+config MODULE_SAUL_DEFAULT
+    select MODULE_SAUL_NRF_TEMPERATURE
+
+choice
+    bool "Backend"
+    depends on MODULE_NETDEV_DEFAULT
+    default NRF802154
+    default NRFBLE
+
+config NRF802154
+    bool "nrf802154"
+    select MODULE_NRF802154
+
+config NRFBLE
+    bool "nrfble"
+    select MODULE_NRFBLE
+
+config NRFMIN
+    bool "nrfmin"
+    select MODULE_NRFMIN
+
+endchoice
```

Comment:
I propose here a couple of changes:
- This should probably be in `cpu/nrf52`
- I think we should define an aux symbol for this radio (e.g NRF52_RADIO) that declares a feature (HAVE_NRF52RF_RADIO). This one can be set to `y` if MODULE_NETDEV_DEFAULT is present. Then, the backend selection depends on HAVE_NRF52_RADIO.

Reply:
>     * This should probably be in `cpu/nrf52`

Ups, yes of course.
 
>     * I think we should define an aux symbol for this radio (e.g NRF52_RADIO) that declares a feature (HAVE_NRF52RF_RADIO). This one can be set to `y` if MODULE_NETDEV_DEFAULT is present. Then, the backend selection depends on HAVE_NRF52_RADIO.

I don't understand why the auxiliary symbol needs to be used seems like a weird dependency. I don't see why `HAVE_NRF52RF_RADIO` would depend on `NETDEV_DEFAULT` either, if its a featuer it should allways be selected. Maybe if you can add a code snipped I can understand better your suggestion.

Reply:
ping @leandrolanzieri I think this was the thing we were talking about.

Reply:
I started #16845 with the HAVE_* but think maybe that is not the best way to move forward.  I think the stage of discussion we are at is:

The boards/cpu  can select things based on other modules enabled (in my case `STDIO`, in your case `NETDEV_DEFAULT`) if selecting is simple and don't have many dependencies or so many options.
The `HAVE_*` is like `FEATURES_PROVIDED` for the features that are not provided in the makefile.

I am still pretty murky of the issues that can arise so it may be good to get together and work through the corner cases.

Reply:
According to what I see in the `Makefile.dep`, the selection of these modules depends on `CPU_MODEL`, I think it might have to do with having `HAS_RADIO_NRF802154` but I'm not sure, as this is done by model name so far.

I think it makes sense to have a symbol like jose proposes, which selects a module that will enable this choice. I'm thinking of something in the direction of:
```
config HAVE_NRF52_RADIO
    bool
    select MODULE_NRF52_RADIO if MODULE_NETDEV_DEFAULT
    help
        Indicates that an NRF52 radio is present.

menuconfig MODULE_NRF52_RADIO
    bool "NRF52 radio driver"
    depends on HAVE_NRF52_RADIO
    depends on TEST_KCONFIG

if MODULE_NRF52_RADIO

choice
    bool "NRF52 radio backend"

config NRF802154
    bool "nrf802154"
    select MODULE_NRF802154

config NRFBLE
    bool "nrfble"
    select MODULE_NRFBLE

config NRFMIN
    bool "nrfmin"
    select MODULE_NRFMIN

endchoice

endif
```